### PR TITLE
updated syntax

### PR DESCRIPTION
**Pull request by Abhiram-Gullapalli**
#### Related issues

Resolves issue #16662 

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by gullapalli9482** [comment]
@wbamberg could you review this if possible

File: files/en-us/web/api/document_object_model/examples/index.md [hunk 1]
```diff
@@ -299,27 +299,27 @@ td { padding: 2px 10px 2px 10px; }
...
   document.getElementById('eventType').innerHTML = e.type;
 
-  var table = document.createElement('table');
-  var thead = table.createTHead();
-  var row = thead.insertRow(-1);
-  var labelList = ['#', 'Property', 'Value'];
-  var len = labelList.length;
+  const table = document.createElement('table');
+  const thead = table.createTHead();
+  const row = thead.insertRow(-1);
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/examples/index.md:311]:
```suggestion
  let row = thead.insertRow(-1);
```

`row` is reassigned at line 323.

File: files/en-us/web/api/document_object_model/examples/index.md [hunk 2]
```diff
@@ -299,27 +299,27 @@ td { padding: 2px 10px 2px 10px; }
...
+  const labelList = ['#', 'Property', 'Value'];
+  const len = labelList.length;
 
-  for (var i=0; i<len; i++) {
+  for (let i=0; i<len; i++) {
     addCell(row, labelList[i]);
   }
 
-  var tbody = document.createElement('tbody');
+  let tbody = document.createElement('tbody');
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/examples/index.md:319]:
```suggestion
  const tbody = document.createElement('tbody');
```

`tbody` is not reassigned.

File: files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 3]
```diff
@@ -31,23 +31,23 @@ This article is an overview of some powerful, fundamental DOM level 1 methods an
...
 
-    for (var j = 0; j < 2; j++) {
+    for (let j = 0; j < 2; j++) {
       // Create a <td> element and a text node, make the text
       // node the contents of the <td>, and put the <td> at
       // the end of the table row
-      var cell = document.createElement("td");
-      var cellText = document.createTextNode("cell in row "+i+", column "+j);
+      const cell = document.createElement("td");
+      let cellText = document.createTextNode("cell in row "+i+", column "+j);
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:50]:
```suggestion
      const cellText = document.createTextNode("cell in row "+i+", column "+j);
```

not reassigned

File: files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 4]
```diff
@@ -269,18 +269,18 @@ The basic steps to create the table in sample1.html are:
 <script>
     function start() {
         // get the reference for the body
-        var mybody = document.getElementsByTagName("body")[0];
+        let mybody = document.getElementsByTagName("body")[0];
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:272]:
```suggestion
        const mybody = document.getElementsByTagName("body")[0];
```

not reassigned

File: files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 5]
```diff
@@ -269,18 +269,18 @@ The basic steps to create the table in sample1.html are:
...
 
         // creates <table> and <tbody> elements
         mytable = document.createElement("table");
         mytablebody = document.createElement("tbody");
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:276]:
```suggestion
        const mytable = document.createElement("table");
        const mytablebody = document.createElement("tbody");
```

File: files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 6]
```diff
@@ -269,18 +269,18 @@ The basic steps to create the table in sample1.html are:
...
 
         // creating all cells
-        for(var j = 0; j < 3; j++) {
+        for(let j = 0; j < 3; j++) {
             // creates a <tr> element
             mycurrent_row = document.createElement("tr");
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:281]:
```suggestion
            const mycurrent_row = document.createElement("tr");
```

File: files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 7]
```diff
@@ -269,18 +269,18 @@ The basic steps to create the table in sample1.html are:
...
 
-            for(var i = 0; i < 4; i++) {
+            for(let i = 0; i < 4; i++) {
                 // creates a <td> element
                 mycurrent_cell = document.createElement("td");
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:285]:
```suggestion
                const mycurrent_cell = document.createElement("td");
```

Comment (wbamberg) [reply files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:285]: Same for `currenttext` at line 287 (I can't make a suggestion for that one).

File: files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 8]
```diff
@@ -355,13 +355,13 @@ Once you have the object in your JavaScript variable, you can set `style` proper
...
        mytable = document.createElement("table");
        mytablebody = document.createElement("tbody");
 
-       for(var row = 0; row < 2; row++) {
+       for(let row = 0; row < 2; row++) {
            mycurrent_row=document.createElement("tr");
-           for(var col = 0; col < 2; col++) {
+           for(let col = 0; col < 2; col++) {
                mycurrent_cell = document.createElement("td");
                currenttext = document.createTextNode("cell is: " + row + col);
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:366]:
```suggestion
               const mycurrent_cell = document.createElement("td");
               const currenttext = document.createTextNode("cell is: " + row + col);
```

File: files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 9]
```diff
@@ -355,13 +355,13 @@ Once you have the object in your JavaScript variable, you can set `style` proper
 </body>
 <script>
     function start() {
-       var mybody = document.getElementsByTagName("body")[0];
+       const mybody = document.getElementsByTagName("body")[0];
        mytable = document.createElement("table");
        mytablebody = document.createElement("tbody");
```
Comment (wbamberg) [inline files/en-us/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:360]:
```suggestion
       const mytable = document.createElement("table");
       const mytablebody = document.createElement("tbody");
```

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks @gullapalli9482 !

In quite a few places you're using `let` where you could use `const`. Remember, the rule is: if the thing is not being reassigned, you can use `const`. "Reassigned" usually means that there is an `=` or `+=` or `-=` (or so on) sign around. For instance:

```js
const myObject = { x: 1; };
// this would be an error
myObject = { y: 1; };
```

...because I'm reassigning the variable. So I would need to use `let`. But this is *not* reassignment:

```js
const myObject = { x: 1; };
// this would NOT be an error
myObject.x = 2;
myObject.y = 3;
```

It's fine to change properties of the object that the variable is pointing to. What's not fine is to make the variable point to a different object.

You're going to have some tricky merge conflicts with this one. Unless you feel confident with Git you may prefer to close this PR and copy your changes into a new one.

**Comment by wbamberg** [comment]
> Thanks @gullapalli9482 !
> 
> In quite a few places you're using `let` where you could use `const`. Remember, the rule is: if the thing is not being reassigned, you can use `const`. "Reassigned" usually means that there is an `=` or `+=` or `-=` (or so on) sign around. For instance:
> 
> ```js
> const myObject = { x: 1; };
> // this would be an error
> myObject = { y: 1; };
> ```
> 
> ...because I'm reassigning the variable. So I would need to use `let`. But this is _not_ reassignment:
> 
> ```js
> const myObject = { x: 1; };
> // this would NOT be an error
> myObject.x = 2;
> myObject.y = 3;
> ```
> 
> It's fine to change members of the object that the variable is pointing to. What's not fine is to make the variable point to a different object.
> 
> You're going to have some tricky merge conflicts with this one. Unless you feel confident with Git you may prefer to close this PR and copy your changes into a new one.

**Pull request closed without merging**